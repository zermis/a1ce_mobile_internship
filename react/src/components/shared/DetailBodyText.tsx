import React from 'react';
import { Text, StyleSheet } from 'react-native';
import Colors from '../../constants/Colors';

interface Props {
    children?: string | number;
    style?: object;
}

const DetailBodyText = (props: Props): JSX.Element => {
    return (
        <Text style={{ ...styles.textBody, ...props.style }}>
            {props.children}
        </Text>
    );
};

const styles = StyleSheet.create({
    textBody: {
        fontFamily: 'ibm-regular',
        fontSize: 14,
        color: Colors.textBody,
        textAlign: 'justify',
        lineHeight: 20
    }
});

export default DetailBodyText;
