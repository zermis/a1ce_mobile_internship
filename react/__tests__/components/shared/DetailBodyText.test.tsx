import DetailBodyText from '../../../src/components/shared/DetailBodyText';
import renderer from 'react-test-renderer';

it('render detail body text correctly', () => {
    const tree = renderer
        .create(<DetailBodyText>{`This is a test`}</DetailBodyText>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
