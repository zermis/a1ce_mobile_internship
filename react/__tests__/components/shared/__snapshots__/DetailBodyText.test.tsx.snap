// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render detail body text correctly 1`] = `
<Text
  style={
    Object {
      "color": "#848484",
      "fontFamily": "ibm-regular",
      "fontSize": 14,
      "lineHeight": 20,
      "textAlign": "justify",
    }
  }
>
  This is a test
</Text>
`;
